src/v/redpanda/admin/services/broker.cc

```diff
@@ -74,4 +85,106 @@ proto::admin::broker broker_service_impl::self_broker() const {
     return b;
 }
 
+namespace {
+
+struct connection_gather_result {
+    chunked_vector<proto::kafka_connection> connections;
+    size_t total_matching_count;
+};
+
+ss::future<connection_gather_result>
+gather_connections(const kafka::server& server, size_t limit) {
+    auto result = connection_gather_result{};
+
+    auto conn_ptrs = server.list_connections();
+    co_await ss::maybe_yield();
+
+    constexpr auto yield_period = 1000;
+    auto i = 0;
+
+    for (const auto& conn_ptr : conn_ptrs) {
+        auto conn_proto = conn_ptr->to_proto();
+
+        // TODO: Apply filtering here when implemented
+        bool matches_filter = true;
+
+        if (matches_filter) {
+            result.total_matching_count++;
+
+            if (result.connections.size() < limit) {
+                result.connections.emplace_back(std::move(conn_proto));
+            }
+        }
+
+        if (++i % yield_period == 0) {
+            co_await ss::maybe_yield();
+        }
+    }
+
+    co_return result;
+}
+
+} // namespace
+
+ss::future<proto::admin::list_kafka_connections_response>
+broker_service_impl::list_kafka_connections(
+  serde::pb::rpc::context ctx,
+  proto::admin::list_kafka_connections_request req) {
+    vlog(brlog.trace, "list_kafka_connections: {}", req);
+
+    // Proxy to the target node id specified in the request
+    auto target = model::node_id{req.get_node_id()};
+    if (target < 0) {
+        throw serde::pb::rpc::invalid_argument_exception(
+          fmt::format("Expected non-negative node id, but got {}", target));
```

Comment:
Other broker APIs support `-1` to be the current node. Let's do that for consistency?